server/blob/blob-cassandra/src/test/java/org/apache/james/blob/cassandra/CassandraBlobsDAOTest.java

```diff
@@ -88,9 +89,24 @@ public void readShouldReturnSplitSavedDataByChunk() throws IOException {
         String longString = Strings.repeat("0123456789\n", MULTIPLE_CHUNK_SIZE);
         BlobId blobId = testee.save(longString.getBytes(StandardCharsets.UTF_8)).join();
 
-        byte[] bytes = testee.read(blobId).join();
+        byte[] bytes = testee.readBytes(blobId).join();
 
         assertThat(new String(bytes, StandardCharsets.UTF_8)).isEqualTo(longString);
     }
 
+    @Test
+    public void readBytesShouldEventualyReturnSameDataAsRead() throws IOException {
+        String longString = Strings.repeat("0123456789\n", MULTIPLE_CHUNK_SIZE);
+        BlobId blobId = testee.save(longString.getBytes(StandardCharsets.UTF_8)).join();
+
+        byte[] bytes = testee.readBytes(blobId).join();
+
+        InputStream read1 = testee.read(blobId);
+        byte[] byteBuffer = new byte[bytes.length];
+        int read = read1.read(byteBuffer);
+        assertThat(bytes.length).isEqualTo(read);
+        assertThat(byteBuffer).isEqualTo(bytes);
+        assertThat(new String(bytes, StandardCharsets.UTF_8)).isEqualTo(longString);
```

Comment:
maybe .hasSameContentAs is simplier to assert on streams.

Reply:
do you mean 
```
assertThat(read1).hasSameContentAs(new ByteArrayInputStream(bytes));
```
?

Reply:
in fact you can even use hasBinaryContent(byte[]) or hasContent(String)